transformer_engine/pytorch/attention.py

```diff
@@ -2753,7 +2781,11 @@ def forward(
         if attn_mask_type == "arbitrary":
             use_flash_attention = False
             use_fused_attention = False
-        if "causal" in attn_mask_type and max_seqlen_q != max_seqlen_kv:
+
+        if (inference_params is None
```

Comment:
Shouldn't your softmax kernel be able to handle it without inference mode as well?